src/server/zset_family.cc

```diff
@@ -2599,6 +2636,296 @@ void ZSetFamily::GeoDist(CmdArgList args, ConnectionContext* cntx) {
                              distance_multiplier);
 }
 
+// Search all eight neighbors + self geohash box
+int membersOfAllNeighbors(ConnectionContext* cntx, string_view key, const GeoHashRadius* n,
+                          GeoShape* shape, GeoArray& ga, unsigned long limit) {
+  GeoHashBits neighbors[9];
+  unsigned int i, last_processed = 0;
+
+  neighbors[0] = n->hash;
+  neighbors[1] = n->neighbors.north;
+  neighbors[2] = n->neighbors.south;
+  neighbors[3] = n->neighbors.east;
+  neighbors[4] = n->neighbors.west;
+  neighbors[5] = n->neighbors.north_east;
+  neighbors[6] = n->neighbors.north_west;
+  neighbors[7] = n->neighbors.south_east;
+  neighbors[8] = n->neighbors.south_west;
+
+  // Get range_specs for neighbors (*and* our own hashbox)
+  std::vector<ZSetFamily::ZRangeSpec> range_specs;
+  for (i = 0; i < sizeof(neighbors) / sizeof(*neighbors); i++) {
+    if (HASHISZERO(neighbors[i])) {
+      continue;
+    }
+
+    // When a huge Radius (in the 5000 km range or more) is used,
+    // adjacent neighbors can be the same, leading to duplicated
+    // elements. Skip every range which is the same as the one
+    // processed previously.
+    if (last_processed && neighbors[i].bits == neighbors[last_processed].bits &&
+        neighbors[i].step == neighbors[last_processed].step) {
+      continue;
+    }
+
+    GeoHashFix52Bits min, max;
+    scoresOfGeoHashBox(neighbors[i], &min, &max);
+
+    ZSetFamily::ScoreInterval si;
+    si.first = ZSetFamily::Bound{static_cast<double>(min), false};
+    si.second = ZSetFamily::Bound{static_cast<double>(max), true};
+
+    range_specs.emplace_back();
+    auto& range_spec = range_specs.back();
+    range_spec.interval = si;
+    ZSetFamily::RangeParams range_params;
+    range_params.interval_type = ZSetFamily::RangeParams::IntervalType::SCORE;
+    range_params.with_scores = true;
+    range_spec.params = range_params;
+
+    last_processed = i;
+  }
+
+  // get all the matching members and add them to the potential result list
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpRanges(range_specs, t->GetOpArgs(shard), key);
+  };
+  OpResult<vector<ScoredArray>> result_arrays =
+      cntx->transaction->ScheduleSingleHopT(std::move(cb));
+  if (result_arrays.status() == OpStatus::WRONG_TYPE) {
+    (*cntx)->SendError(kWrongTypeErr);
+    return -1;
+  }
+
+  // filter potential result list
+  unsigned long count = 0;
+  double xy[2];
+  double distance;
+  for (auto& arr : *result_arrays) {
+    for (auto& p : arr) {
+      if (geoWithinShape(shape, p.second, xy, &distance) == true) {
```

Comment:
From reading `geoWithinShape` comment, it seems like it never returns `true`. It may return `C_OK` on success, or `C_ERR` on error.
I suspect that this is a bug, could you add also a test case to catch it before fixing?